eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken
+
+## Clone the repo from git
+GitHub-Clone $targetRepoName
+
+## Create a branch from the repo with the given SHA. 
+$targetBranch = "dev/" + $githubUser + "/arcade-" + $arcadeSdkVersion
+$githubUri = Get-Github-RepoAuthUri $targetRepoName
+
+$darcBranchName = "refs/heads/" + $targetBranch
+$darcRepoName = "https://github.com/${githubOrg}/${targetRepoName}"
+
+# check to see if branch exists and clean it up if it does
+$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+if($null -ne $branchExists)
+{
+    & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+    Git-Command $targetRepoName push origin --delete $targetBranch
+}
+Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Make the changes to that branch to update Arcade - use darc
+Set-Location $(Get-Repo-Location $targetRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+
+## Push branch to github
+Git-Command $targetRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
+Git-Command $targetRepoName push origin HEAD
+
+## Push branch to AzDO org/project with the official pipeline to build the repo
+## Don't need to do this for Roslyn, but we'll need to do it for Runtime. 
+
+## Add default channel from that github repo and branch to "General Testing"
+& darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+
+## Run an official build of the branch using the official pipeline
+$buildId = Invoke-AzDOBuild -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken -branchName $targetBranch -buildParameters $buildParameters
+
+## Check build for completion every 5 minutes. 
+while("completed" -ne (Get-BuildStatus -azdoOrg $azdoOrg -azdoProject $azdoProject -buildId $buildId -azdoToken $azdoToken))
+{
+    Write-Host "Waiting for build to complete..."
+    Start-Sleep -Seconds (5*60)
+}
+
+## Output summary of references for investigations
+Write-Host "Arcade Version: ${arcadeSdkVersion}"
+Write-Host "Repository Cloned: ${githubOrg}/${targetRepoName}"
+Write-Host "Branch name in repository: ${targetBranch}"
+Write-Host "Last Known Good build SHA: ${sha}"
+
+## If build fails, then exit
+if("failed" -eq (Get-BuildResult -azdoOrg $azdoOrg -azdoProject $azdoProject -buildId $buildId -azdoToken $azdoToken))
+{
+    Write-Error "Build failed"
+    exit
+}
+
+## Clean up branch if successful
```

Comment:
Don't we need to run these when things fail? Perhaps would be useful to create a "finally {...} " block to make sure a clean up is always done.

Reply:
I thought it might be useful to keep the branch around for investigation purposes if there is a failure. @mmitche do you have an opinion on whether keeping these around would be useful or not? (I have an issue to write a clean up script when we're finished with investigations).

Reply:
I'm going to resolve this so I can merge this PR. If we decide to always clean up because we don't need the branches for investigation, I'm happy to change it :)